torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -773,6 +773,12 @@ Tensor trace_backward(const Tensor & grad, IntArrayRef sizes) {
   return grad_input.view(sizes);
 }
 
+Tensor sgn_backward(Tensor result, Tensor grad, Tensor self) {
+  // [grad / abs(z) - Re(grad/self) * result
```

Comment:
For a real Tensor this should always return 0 right? That does not seem to be the case here when self < 0 right?